neural_modelling/src/neuron/c_main_neurons.c

```diff
@@ -0,0 +1,328 @@
+/*
```

Comment:
These are actually purposefully different because they don't refer to matching things.  The c_main_neuron.h header is used in both c_main and c_main_neurons.c and similarly c_main_synapse.h is used in both c_main and c_main_synapses.c.  I will however rename them to make it less confusing!

Reply:
Renamed now to add "common" to the end